src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,112 @@
+package duke.parser;
+
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.EventCommand;
+import duke.command.FindCommand;
+import duke.command.IncorrectCommand;
+import duke.command.ListCommand;
+import duke.command.ToDoCommand;
+import duke.exception.DukeException;
+import duke.util.DateUtil;
+
+import java.util.Date;
+
+/**
+ * Represents a parser that parses strings and creates commands.
+ */
+public class Parser {
+
+    /**
+     * Parses the command contained in a {@link String} and returns a {@link Command} that represents it.
+     *
+     * @param fullCommand {@link String} containing the command.
+     * @return {@link Command} representing the command.
+     * @throws DukeException if the command has parser errors.
+     */
+    public static Command parse(String fullCommand) throws DukeException {
+        String[] arr = fullCommand.trim().split(" ", 2);
+        String commandWord = arr[0];
+
+        try {
+            switch (commandWord) {
+            case DeadlineCommand.COMMAND_WORD:
```

Comment:
https://stackoverflow.com/questions/9969690/whats-the-advantage-of-a-java-enum-versus-a-class-with-public-static-final-fiel
There’s an entire discussion on this here - it is true that enums have advantages, but I don’t really need them in this context. I wanted to keep all the relevant information about a Command in its implemented class, but I agree with your points.
On hindsight, using enums will allow me to point other command words to a Command - but I’m not sure if that’s a good thing to have since it can be misleading.